controllers/replica_set_controller.go

```diff
@@ -182,22 +178,6 @@ func (r ReplicaSetReconciler) Reconcile(ctx context.Context, request reconcile.R
 		)
 	}
 
-	// if we're scaling down, we need to wait until the StatefulSet is at the
-	// desired number of replicas. Scaling down has to happen one member at a time
-	if scale.IsScalingDown(mdb) {
```

Comment:
the check for scaling down no longer needs to happen as the readiness check later on accepts the expected number of replicas that should be ready.